packages/browser/src/Session.ts

```diff
@@ -132,16 +134,49 @@ function isLoggedIn(
 /**
  * A {@link Session} object represents a user's session on an application. The session holds state, as it stores information enabling acces to private resources after login for instance.
  */
-export class Session extends EventEmitter {
+export class Session extends EventEmitter implements IHasSessionEventEmitter {
   /**
    * Information regarding the current session.
    */
   public readonly info: ISessionInfo;
 
+  /**
+   * Session attribute exposing the EventEmitter interface, to listen on session
+   * events such as login, logout, etc.
+   */
+  public readonly events: ISessionEventEmitter;
+
   private clientAuthentication: ClientAuthentication;
 
   private tokenRequestInProgress = false;
 
+  /**
+   * Proxy handler to ease transition to the Session no longer being an EvenEmitter,
+   * and instead Session["events"] being a SessionEventEmitter. For the time being,
+   * Session["events"] is just a proxy to the SessionEventEmitter interface of
+   * the Session.
+   *
+   * When Session no longer implements SessionEventEmitter, this can be removed.
+   * @hidden
+   */
+  private handler = {
+    // This proxy is only a temporary measure until Session no longer extends
+    // SessionEventEmitter, and the proxying is no longer necessary.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    get(target: any, prop: any, receiver: any) {
+      // Reject any calls to the proxy that isn't specific to the EventEmitter API
+      if (
+        !Object.getOwnPropertyNames(EventEmitter).includes(prop) &&
+        Object.getOwnPropertyNames(Session.prototype).includes(prop)
+      ) {
+        throw new Error(
+          `events only implements SessionEventEmitter, [${prop}] is not supported`
+        );
+      }
+      return Reflect.get(target, prop, receiver);
+    },
+  };
```

Comment:
This handler can just be an object exported from the authn-core packages and re-used by the browser + node packages.

Reply:
Good idea, done in 891c1b83